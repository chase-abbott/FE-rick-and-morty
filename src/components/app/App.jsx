import React from 'react';
import { Switch, Route } from 'react-router-dom';
import Auth from './auth/Auth';
import Favorites from './Favorites';
import styled from 'styled-components';
import Home from './Home';

export default function App() {
  return (
    <Body>
      <Switch>
        <Route exact path="/" component={Auth}/>
        <Route exact path="/home" component={Home}/>
        <Route exact path="/favorites" component={Favorites}/>
      </Switch>
    </Body>
  );
}

const Body = styled.div`
width: 100%;
height: 100%;
display: flex;
justify-content: center;
 `;
