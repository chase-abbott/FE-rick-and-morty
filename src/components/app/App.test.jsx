import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { MemoryRouter as Router } from 'react-router';
import App from './App';

describe('App component', () => {
  afterEach(() => cleanup());
  it('renders App', () => {
    const { asFragment } = render(<Router><App /></Router>);
    expect(asFragment()).toMatchSnapshot();
  });
});
