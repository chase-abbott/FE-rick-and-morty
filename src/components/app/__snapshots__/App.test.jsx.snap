// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders App 1`] = `
<DocumentFragment>
  <div
    class="sc-dIsUp BDQMc"
  >
    <form
      style="width: 50%;"
    >
      <h1>
         Pick Your Rick 
      </h1>
      <div
        class="sc-bdnxRM ebmJOv"
      >
        <div>
          <label
            for="username"
          >
             Username 
          </label>
          <input
            class="sc-gtsrHT ScCSh"
            name="username"
            type="text"
            value=""
          />
        </div>
        <div>
          <label
            for="password"
          >
             Password 
          </label>
          <input
            class="sc-gtsrHT ScCSh"
            name="password"
            type="password"
            value=""
          />
        </div>
        <div
          class="sc-dlnjwi hBfhAE"
        >
          <button
            class="sc-hKFxyN"
            value="login"
          >
             Log In 
          </button>
          <button
            class="sc-hKFxyN"
            value="signup"
          >
             Sign Up 
          </button>
        </div>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
